lidar_prod/tasks/building_validation.py

```diff
@@ -326,26 +344,32 @@ def _make_detailed_group_decision(
         # REFUTATION
         ia_refuted = (
             np.mean(
-                (1 - infos.probabilities) >= self.thresholds.min_confidence_refutation
+                (1 - infos.probabilities)
+                >= self.thresholds.min_confidence_refutation
             )
             >= self.thresholds.min_frac_refutation
         )
         uni_overlayed = (
             np.mean(infos.overlays) >= self.thresholds.min_uni_db_overlay_frac
         )
+        # If low entropy, we may trust AI to confirm/refute
+        if not high_entropy:
+            if ia_refuted:
+                if uni_overlayed:
+                    return self.codes.detailed.ia_refuted_but_under_db_uni
+                return self.codes.detailed.ia_refuted
+            if ia_confirmed:
+                if uni_overlayed:
+                    return self.codes.detailed.both_confirmed
+                return self.codes.detailed.ia_confirmed_only
+        # Else, we may still use BDUni information
+        if uni_overlayed:
+            return self.codes.detailed.db_overlayed_only
 
+        # Else: we are uncertain, and we specify why we can specify if entropy was
+        # involved to conclude to uncertainty.
         if high_entropy:
             return self.codes.detailed.unsure_by_entropy
-        if ia_refuted:
-            if uni_overlayed:
-                return self.codes.detailed.ia_refuted_but_under_db_uni
-            return self.codes.detailed.ia_refuted
-        if ia_confirmed:
-            if uni_overlayed:
-                return self.codes.detailed.both_confirmed
-            return self.codes.detailed.ia_confirmed_only
-        if uni_overlayed:
-            return self.codes.detailed.db_overlayed_only
         return self.codes.detailed.both_unsure
```

Comment:
On ne passe dans cette ligne (return self.codes.detailed.both_unsure) que si on est à "not high_entropy", je suggère de la passer dans le bloc "if" correspondant, en ligne 365.

Eventuellement en plus, en traitant le cas "high entropy" en premier et en finissant sur le bloc "if not high_entropy" il est possible de supprimer la ligne du "if", puisqu'on est sorti de la méthode en cas de "high entrpoy" just avant. Cela fait une indentation de moins dans un bloc qui en a beaucoup (c'est juste une question de lisibilité)